linalg/sparsemat.cpp

```diff
@@ -2473,7 +2473,7 @@ void SparseMatrix::AddSubMatrix(const Array<int> &rows, const Array<int> &cols,
          {
             // if the element is zero do not assemble it unless this breaks
```

Comment:
Can this comment be updated to reflect the modified conditional statement, maybe something like "skip assembly of zero elements if either (i) symmetry is not broken or (ii) user specified to skip zeros regardless of symmetry"?

Sorry, I should have caught this on my last review

Reply:
Updated comment!